server/sonar-web/src/main/js/app/components/nav/component/ComponentNavBranchesMenuItem.tsx

```diff
@@ -46,7 +46,7 @@ export default function ComponentNavBranchesMenuItem({ branch, ...props }: Props
           active: props.selected
         })}
         to={getProjectBranchUrl(props.component.key, branch)}>
-        <div className="navbar-context-meta-branch-menu-item-name">
+        <div className="navbar-context-meta-branch-menu-item-name text-ellipsis">
```

Comment:
Maybe a tooltip should be added for branches with a name that are truncated (longer than x characters) ?